iree/turbine/kernel/wave/runtime/runtime.cpp

```diff
@@ -101,26 +86,40 @@ int launch(const KernelLaunchInfo &info, const Int64Vector &tensors,
         HIP_LAUNCH_PARAM_BUFFER_SIZE,
         &kernArgSize,
         HIP_LAUNCH_PARAM_END};
+
     HIP_CHECK_RETURN(hipModuleLaunchKernel(function, info.gridX, info.gridY, info.gridZ,
                                            info.blockX, info.blockY, info.blockZ, info.sharedMemoryBytes,
                                            stream, nullptr, (void **)&hipLaunchParams));
 
-    // Always unload the module if the kernel hash is empty.
-    if (info.kernelHash.empty())
-        HIP_CHECK_RETURN(hipModuleUnload(module));
-
     return hipSuccess;
 }
 
+static void unload_binary(void *ptr) noexcept
+{
+    auto module = reinterpret_cast<hipModule_t>(ptr);
+
+    // Assume it will never fail as we cant do anything meaningful about it
+    // anyway.
+    (void)hipModuleUnload(module);
```

Comment:
Can we add a break or even a print saying that the `free` was not successful if that's the case?

Reply:
done